### PR TITLE
chore(nix): remove the opam2nix dependency on the default package

### PR DESCRIPTION
**Pull request by anmonteiro**
I followed the derivation here: https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/tools/ocaml/dune/3.nix

Inspired by the changes in https://github.com/ocaml/ocaml-lsp/pull/1040

### CODE REVIEW

**Comment by anmonteiro** [comment]
To try this out: 

```
$ nix run github:anmonteiro/dune/anmonteiro/nix-package-default# -- --version
```

File: flake.nix [hunk 1]
```diff
@@ -88,7 +88,24 @@
...
     in
     {
-      packages.default = scope.dune;
+      packages = {
+        dune = scope.dune;
+        default = with pkgs; stdenv.mkDerivation rec {
+          pname = package;
+          version = "n/a";
+          src = ./.;
+          nativeBuildInputs = with ocamlPackages; [ ocaml findlib ];
```
Comment (rgrinberg) [inline flake.nix:97]: Why do we need findlib? We don't use any OCaml libraries. Or is it for the installation step?

Comment (anmonteiro) [reply flake.nix:97]: Yup, findlib has a `setupHook` ([source](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/tools/ocaml/findlib/default.nix#L30)) that sets the `OCAMLFIND_DESTDIR` variable.

Comment (Alizter) [reply flake.nix:97]: Btw if you ever need to use opam inside this flake, be aware that `OCAMLFIND_DESTDIR` will break package builds in opam so make sure to unset it.

**Pull request merged**